Model/Checkout/PlaceOrder.php

```diff
@@ -238,15 +238,17 @@ public function createOrder(Quote $quote, string $amwalOrderId, string $refId):
         $this->logDebug(sprintf('Submitting quote with ID %s', $quote->getId()));
         $order = $this->getOrderByAmwalOrderId($amwalOrderId);
 
-        if ($order->getEntityId()) {
-            if ($order->getState() == Order::STATE_PENDING_PAYMENT) {
-                $this->orderRepository->delete($order);
-                $order = $this->quoteManagement->submit($quote);
-            }
-        }else{
-            $order = $this->quoteManagement->submit($quote);
+        if ($order) {
+            $this->logDebug(
+                sprintf('Existing order with ID %s found. Canceling order and re-submitting quote.', $order->getEntityId())
+            );
+            $order->cancel();
+            $order->setAmwalOrderId($amwalOrderId . '-canceled');
```

Comment:
with this change we need to change the cron job to safely handle `$amwalOrderId . '-canceled'`

Reply:
I added this now, I implemented it so it would skip orders which have the "-canceled" suffix. The reason behind this is that now if the first order is canceled, and the second is successful (with the same amwal order id) we would otherwise trigger an order update for the canceled order, even though there already is a successful order.